src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesGcm.Unix.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Internal.Cryptography;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Security.Cryptography
+{
+    partial class AesGcm
+    {
+        // See: aes_gcm_ctrl: conditions related to EVP_CTRL_GCM_SET_TAG/EVP_CTRL_GCM_GET_TAG
+        public static KeySizes TagByteSizes { get; } = new KeySizes(1, 16, 1);
+        private SafeEvpCipherCtxHandle _ctxHandle;
+
+        public AesGcm(ReadOnlySpan<byte> key)
+        {
+            if (key == null)
```

Comment:
As a `null` check it should be on the array one.  I'm still not sure what this is comparing against, but it probably isn't the right thing.